daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -620,6 +641,26 @@ object Speedy {
     def execute(v: SValue, machine: Machine): Unit
   }
 
+  /*
+   Speedy uses a caller-saves strategy for managing the environment.  In a Speedy machine,
```

Comment:
Wouldn't it make more sense to put that comment above the definition of the `Machine` class? It might also help to give this comment some sort of label that can be referenced from other comments and then easily be found.

Reply:
done